File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/InArray.java
package com.onevgo.functions;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.Map;

public class InArray {
    public static boolean inArray(Object needle, Collection haystack) {
        return haystack.contains(needle);
    }

    public static boolean inArray(Object needle, Map haystack) {
        return haystack.containsValue(needle);
    }

    public static void main(String[] args) {
        List<String> os = Arrays.asList("Mac", "NT", "Irix", "Linux");
        System.out.println(inArray("Irix", os));
        System.out.println(inArray("mac", os));
    }
}

File: 01_Language/01_Functions/js/in_array.js
function in_array(needle, haystack, strict = false) {
    for (let k in haystack) {
        if (strict && haystack[k] === needle) {
            return true
        } else if (!strict && haystack[k] == needle) {
            return true
        }
    }
    return false
}
